File: src/main/java/io/stouder/adonis/service/AdonisAppService.java
package io.stouder.adonis.service;

import com.intellij.openapi.project.Project;

import java.util.List;

public interface AdonisAppService {
    static AdonisAppService getInstance(Project project) {
        return project.getService(AdonisAppService.class);
    }

    boolean isAdonisProject();

    List<String> getAdonisRoots();
}

File: src/main/java/io/stouder/adonis/service/AdonisAppService.kt
package io.stouder.adonis.service

import com.intellij.openapi.project.Project

interface AdonisAppService {
    companion object {
        fun getInstance(project: Project): AdonisAppService {
            return project.getService(AdonisAppService::class.java)
        }
    }

    fun isAdonisProject(): Boolean

    fun getAdonisRoots(): List<String>
}